client/features/output-visualization-page/components/UndoAndRedo.tsx

```diff
@@ -16,38 +17,26 @@ const useTemporalStore = <T,>(
 
 const UndoAndRedo = () => {
   const { undo, redo } = useTemporalStore((state) => state);
```

Comment:
We can destructure undo, redo, pastStates, and futureStates from the zustand store in a single call. This will reduce redundant calls to the store. The example code will look like:
```typescript
  const { undo, redo, pastStates, futureStates } = useUndoRedoStore(
    (state) => ({
      undo: state.undo,
      redo: state.redo,
      pastStates: state.pastStates,
      futureStates: state.futureStates,
    }),
  );

  const canUndo = pastStates.length > 0;
  const canRedo = futureStates.length > 0;
```